python/dasclient/dasclient.py

```diff
@@ -141,14 +139,19 @@ def _get(self, path, stream=False, **kwargs):
         self.logger.debug("Fail: " + response.text)
         raise DasClientException('Failed to call DAS web service.')
 
-        
-    def _post(self, path, payload):
+    
+    def _call(self, path, payload, method):
         headers = {'Content-Type': 'application/json',
                    'User-Agent': self.user_agent}
         headers.update(self.auth_headers())
         body = json.dumps(payload)
-        
-        response = requests.post(self._das_url(path), data=body, headers=headers)
+
+        response = None
+        if(method == "POST"):
+            response = requests.post(self._das_url(path), data=body, headers=headers)
+        elif(method == "PATCH"):
+            response = requests.patch(self._das_url(path), data=body, headers=headers)
+            
```

Comment:
Here you can avoid some extra steps by passing `json=payload` (instead of data=) in calls to post and patch. This will automatically marshal the data to json, as well as add the right content-type header.

And the response can get `response.json()['data']` 


A bonus (and this might be nitpicky) would be to resolve your method to a reference and have one line that calls it (again to eliminate redundant code), like so:

```
    ...
    fmap = {'POST': requests.post, 'PATCH': requests.patch}
    try:
        fn = fmap[method]
    except KeyError:
        self.logger.error('method must be one of...')
    else:
        response = fn(self._das_url(path), data=body, headers=headers)
```